subsys/bluetooth/host/gatt.c

```diff
@@ -2011,44 +2013,46 @@ static u8_t update_ccc(const struct bt_gatt_attr *attr, void *user_data)
 
 	ccc = attr->user_data;
 
-	for (i = 0; i < ARRAY_SIZE(ccc->cfg); i++) {
-		/* Ignore configuration for different peer */
-		if (bt_conn_addr_le_cmp(conn, &ccc->cfg[i].peer)) {
-			continue;
+	/* Check if attribute requires encryption/authentication */
+	need_update = false;
+	err = bt_gatt_check_perm(conn, attr, BT_GATT_PERM_WRITE_MASK);
+	if (err) {
+		if (err == BT_ATT_ERR_WRITE_NOT_PERMITTED) {
+			BT_WARN("CCC %p not writable", attr);
+			return BT_GATT_ITER_CONTINUE;
 		}
-
-		/* Check if attribute requires encryption/authentication */
-		err = bt_gatt_check_perm(conn, attr, BT_GATT_PERM_WRITE_MASK);
-		if (err) {
-			bt_security_t sec;
-
-			if (err == BT_ATT_ERR_WRITE_NOT_PERMITTED) {
-				BT_WARN("CCC %p not writable", attr);
-				continue;
-			}
-
+		if (IS_ENABLED(CONFIG_BT_SMP)) {
 			sec = BT_SECURITY_L2;
-
 			if (err == BT_ATT_ERR_AUTHENTICATION) {
 				sec = BT_SECURITY_L3;
 			}
-
 			/* Check if current security is enough */
-			if (IS_ENABLED(CONFIG_BT_SMP) &&
-			    bt_conn_get_security(conn) < sec) {
-				if (data->sec < sec) {
-					data->sec = sec;
-				}
-				continue;
-			}
+			need_update = bt_conn_get_security(conn) < sec &&
```

Comment:
I think this should be:

```
		    	if (bt_conn_get_security(conn) < sec)) {
				data->sec = sec;
				return BT_GATT_ITER_CONTINUE;
			}
```
If not then please explain why I'm wrong because I'm currently not seeing it.

Reply:
At this point, we do not know whether the update was triggered by an unrelated peer, that check happens below. If we update `data->sec` here, we would update for _any_ connection.